cxbox-core/src/main/java/org/cxbox/core/crudma/bc/BusinessComponent.java

```diff
@@ -77,7 +78,11 @@ public Long getIdAsLong() {
 	}
 
 	public Long getParentIdAsLong() {
-		return getAsLong(parentId);
+		try {
+			return getAsLong(parentId);
+		} catch (NumberFormatException numberFormatException) {
+			throw new EntityNotFoundException();
```

Comment:
throw new EntityNotFoundException(this.typeOfEntity.getSimpleName(), bc.getIdAsLong()) replace with EntityNotFoundException(this.typeOfEntity.getSimpleName(), bc.getId())

add new corresponding methods to public class EntityNotFoundException